elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/Cardinality.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates whether a dimension has small, medium, or large cardinality.
+ * <p>
+ * Example: {@literal @}Cardinality(size = {@link DimensionSize#MEDIUM}). If {@code size} is not specified,
+ * {@link DimensionSize#LARGE} is used by default. See {@link DimensionSize}.
+ * <p>
+ * To avoid double binding, {@link Cardinality} is a {@link ElementType#TYPE class-only annotation}.
+ */
+@Documented
+@Target(ElementType.TYPE)
```

Comment:
I make Cardinality a class-level only annotation because I though it would be awkward to have a class being annotated and a second person put the same annotation in a relationship again.

Reply:
Right. Based on your other comment, I'll also add `ElementType.FIELD` and `ElementType.METHOD`